### ISSUE TITLE
Running react in development mode

### ISSUE THREAD

**Issue by aymericbouzy** [netguru/react_webpack_rails#127]
When trying to use `react-addons-perf`, and specifically when I call `Perf.start()`, I get the following error : 

```
react_bundle.self-6e16580….js?body=1:143 ReactPerf is not supported in the production builds of React. To collect measurements, please use the development build of React instead.
```

Command `$ echo $NODE_ENV` returns an empty string, and I'm running `npm run start-hot-dev`.
Thanks for your help.

**Comment by mxaly** [comment]
hi @aymericbouzy 
looks like we should specify development env as well. For now you can do it in webpack's dev config. Simply add: 
```
config.plugins.push(
  new Webpack.DefinePlugin({'process.env': {'NODE_ENV': '"production"'}})
);
```
to your `webpack/dev.config.js`

**Comment by mxaly** [comment]
@aymericbouzy i was not able to reproduce that - perf was working fine on my development (running webpack server in dev/hot reload mode) 
Keep in mind that Perf is made to work on development and running `npm run build` is creating production build for you. IE you should use it with webpack dev server (`npm start`)

Anyway, we have added NODE_ENV=development to generated webpack cofig at dev env for consistency.

**Comment by aymericbouzy** [comment]
Sort of reopening this issue : I understand better now what went wrong. I had indeed run `npm run build` which had created a `react_bundle.js` file which was served through my asset pipeline. Therefore, in dev mode, I had to React instances running (not advisable ...)

Here is my solution : 
creating `app/assets/javascripts/conditional_react_bundle.js.erb` reading : 
```
<% require_asset 'react_bundle' unless Rails.env.development? %>
```
and replacing `//= require react_bundle` with `require conditional_react_bundle` in `application.js`.

I'd be glad to know what you think of my solution and hope it will be useful to others!
